### PR TITLE
Query caching

### PR DESCRIPTION
**Pull request by nottoday2471**

### CODE REVIEW

File: src/pages/orders/Orders.tsx [hunk 1]
```diff
@@ -20,41 +21,61 @@ interface IGetOrders {
...
-      })
-      .then((data: IOrderResponse) => {
-        setOrders(data.data);
-        setTotalPages(data.totalPages);
-        return data;
-      })
-      .catch(err => {
-        setError(err);
-        setOrders([]);
+  const { isLoading, data, error, isError } = useQuery<IOrder[], AxiosError>(
```
Comment (RamboUA) [inline src/pages/orders/Orders.tsx:31]: I think it is better to make as a hook, for being able to use it in multiple places

**Review by RamboUA — CHANGES_REQUESTED** [review]
Great job. Just check a comment please

File: src/hooks/orders.hook.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      {
+        page,
+        limit,
+        statusFilter: selectedFilter,
+        sortBy: sortValue,
+        sort: sortType,
+      },
+    ],
+    async () => {
+      const orders = await fn({
```
Comment (BillyAleks) [inline src/hooks/orders.hook.ts:21]: you could use here directly getOrders, as it should be the only possible query for this hook.

File: src/helpers/getOrders.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+) => {
+  const res = await orderService.getOrders({
+    page: config.page,
+    limit: config.limit || 10,
+    status: config.statusFilter,
+    sortBy: config.sortBy,
+    sort: config.sort,
+  });
+
+  setTotalPages(res.totalPages);
```
Comment (BillyAleks) [inline src/helpers/getOrders.ts:16]: It's not really good idea to break single responsibility principle of function. Better to bring a result outside and set pages count on that level

**Review by BillyAleks — COMMENTED** [review]
So approximate flow should be like this:
- You have hook, which just making a bind service for useQuery;
- Inside of this hook you use query getOrders;
- in Orders.tsx you get your { isLoading, data, error, isError }
- also here you should use useEffect to set pages count

**Pull request merged**